src/lib/net/address.c

```diff
@@ -852,6 +852,24 @@ tor_addr_is_valid_ipv4n(uint32_t v4n_addr, int for_listening)
   return v4n_addr != 0;
 }
 
+/* IPv6 equivalent for tor_addr_is_valid() */
+int
+tor_addr_ipv6_is_valid(const tor_addr_t *addr)
```

Comment:
Hi, you don't need a separate tor_addr_is_valid() function for IPv6. tor_addr_is_valid() already has enough IPv6 to do what you need.

You can just say: "tor_addr_is_valid(ipv6_addr, 0)".